osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1309 @@
+}
+
+/// Gets a ipmi_lanparm_t*.  Does all registration needed with IPMIClient.
+void getPARM(ipmi_lanparm_t* lp,
+             unsigned int parm,
+             const std::string key,
+             const std::string colName,
+             IPMIClient* client) {
+  parmData* data = new parmData;
+  data->client = client;
+  data->mcKey = key;
+  data->colName = colName;
+  data->parm = parm;
+  client->pushParmData(data);
+
+  int rv = ipmi_lanparm_get_parm(
+      lp,
+      parm,
+      0,
+      0,
+      [](ipmi_lanparm_t* lanparm,
+         int err,
+         unsigned char* parmVal,
+         unsigned int dataLen,
+         void* retData) {
+
+        parmData* d = (parmData*)retData;
+        std::unique_ptr<parmData, std::function<void(parmData*)>> freeParm(
+            d, [](parmData* parm) {
+              TLOG << "Freeing OpenIPMI LANParm from unique_ptr";
+              parm->client->rmParmData(parm);
+            });
+
+        if (err != 0) {
+          TLOG << "Unexpected error while getting PARM value: "
+               << strerror(err);
+          return;
+        }
+        // If there is an error, unlocking will also error out.
+        std::unique_ptr<ipmi_lanparm_t, std::function<void(ipmi_lanparm_t*)>>
+            unlock(lanparm, unlockLANPARM);
+
+        switch (d->parm) {
+        case IPMI_LANPARM_BACKUP_GATEWAY_ADDR:
+        case IPMI_LANPARM_IP_ADDRESS:
+        case IPMI_LANPARM_DEFAULT_GATEWAY_ADDR:
+        case IPMI_LANPARM_SUBNET_MASK:
+          d->client->insertRowsQueue(
+              d->mcKey, d->colName, toCIDR(parmVal, dataLen));
+          break;
+
+        case IPMI_LANPARM_BACKUP_GATEWAY_MAC_ADDR:
+        case IPMI_LANPARM_DEFAULT_GATEWAY_MAC_ADDR:
+        case IPMI_LANPARM_MAC_ADDRESS:
+          d->client->insertRowsQueue(
+              d->mcKey, d->colName, toMAC(parmVal, dataLen));
+          break;
+
+        case IPMI_LANPARM_IP_ADDRESS_SRC:
+          d->client->insertRowsQueue(
+              d->mcKey, d->colName, toAddressSrc(parmVal, dataLen));
+          break;
+
+        case IPMI_LANPARM_COMMUNITY_STRING: {
+          std::string comm(reinterpret_cast<const char*>(parmVal + 1),
+                           static_cast<size_t>(dataLen));
+          d->client->insertRowsQueue(d->mcKey, d->colName, comm);
+          break;
+        }
+
+        case IPMI_LANPARM_SET_IN_PROGRESS:
+          d->client->insertRowsQueue(
+              d->mcKey, d->colName, toSetInProgress(parmVal, dataLen));
+          break;
+
+        default:
+          LOG(ERROR) << "Got an unexpected LANPARM type: "
+                     << std::to_string(d->parm);
+          break;
+        }
+
+      },
+      data);
+
+  if (rv != 0) {
+    LOG(ERROR) << "Could not get default gateway lanparm: " << strerror(rv);
+  }
+}
+
+/*===========================OpenIPMI Callbacks================================
+ */
+
+void getLANsCB(ipmi_domain_t* domain, ipmi_mc_t* mc, void* data) {
+  IPMIClient* client = (IPMIClient*)data;
```

Comment:
Try and avoid C-style casts if you can. This might require a `reinterpret_cast<IMPIClient*>` unfortunately, but I believe we'd rather use a `reinterpret_cast` as opposed to C-styel casts.